File: ReactMXHApi6/ReactMXHApi6/Helper/Pagination.cs
namespace ReactMXHApi6.Helper
{
    public class Pagination<T> where T : class
    {
        public Pagination(int pageNumber, int pageSize, int count, IReadOnlyList<T> data)
        {
            PageNumber = pageNumber;
            PageSize = pageSize;
            Count = count;
            TotalPages = (int)Math.Ceiling(count / (double)pageSize);
            Data = data;
        }
        public int TotalPages { get; set; }
        public int PageNumber { get; set; }
        public int PageSize { get; set; }
        public int Count { get; set; }
        public IReadOnlyList<T> Data { get; set; }
    }
}

File: ReactClient/src/models/pagination.ts
export interface IPagination {
    pageNumber: number;
    pageSize: number;
    count: number;
    totalPages: number;
}

export class Pagination implements IPagination{
    pageNumber: number;
    pageSize: number;
    count: number;
    totalPages: number;

    constructor(pageNumber: number, pageSize: number, count: number, totalPages: number){
        this.pageNumber = pageNumber;
        this.pageSize = pageSize;
        this.count = count
        this.totalPages = totalPages;
    }
}

export class PaginatedResult<T> {
    data: T;
    pageNumber: number;
    pageSize: number;
    count: number;
    totalPages: number;

    // pagination: IPagination;

    // constructor(data: T, pagination: IPagination) {
    //     this.data = data;
    //     this.pagination = pagination;
    // }

    constructor(pageNumber:number, pageSize: number, count: number, totalPages: number, data: T) {
        this.data = data;
        this.pageNumber = pageNumber;
        this.pageSize = pageSize;
        this.count = count
        this.totalPages = totalPages;
    }
}

export class PagingParams {
    pageNumber;
    pageSize;

    constructor(pageNumber = 1, pageSize = 5) {
        this.pageNumber = pageNumber;
        this.pageSize = pageSize;
    }
}